packages/editor-ui/src/plugins/i18n/index.ts

```diff
@@ -0,0 +1,330 @@
+import _Vue from "vue";
+import axios from 'axios';
+import VueI18n from 'vue-i18n';
+import { Store } from "vuex";
+import Vue from 'vue';
+import { INodeTranslationHeaders, IRootState } from '@/Interface';
+const englishBaseText = require('./locales/en');
+
+Vue.use(VueI18n);
+
+const REUSABLE_DYNAMIC_TEXT_KEY = 'reusableDynamicText';
+const CREDENTIALS_MODAL_KEY = 'credentialsModal';
+const NODE_VIEW_KEY = 'nodeView';
+
+export function I18nPlugin(vue: typeof _Vue, store: Store<IRootState>): void {
+	const i18n = new I18nClass(store);
+
+	Object.defineProperty(vue, '$i', {
+		get() { return i18n; },
+	});
+
+	Object.defineProperty(vue.prototype, '$i', {
+		get() { return i18n; },
+	});
```

Comment:
I know `$i` keeps the code very concise, but it is overly used in loops etc. as well which strips ability to derive it.

options I would maybe also consider in order would be `$translate`/`$translation` or `$locale`

wdyt?

Reply:
I chose `$i` by ruling out other options, but I agree/hope there might be something better.

`translation` would be inaccurate because all strings, including untranslated English strings, rely on this method as well. This is why the mixin (before removal) was called `renderText`.

`$locale` as a plugin could be mixed up with `locale` as an identifier, i.e. `en`, `de`, etc. Not too bad, but we'd using the same word for entirely different things.

With Mutasem we considered `I18N`, but the caps look very out of place and we're trained to think of caps as a constant instead of a namespace.

Looping in `<template>` usually doesn't expose the index and when it does, Vue docs show it's to be spelled out as `index`. Looping in `<script>` does use `i` but I find them hard to mix up because `$i` will always be part of `this.$i.method()`.